### DISCUSSION TITLE
Moving between platforms

### DISCUSSION THREAD

**Discussion by laurel-nave-powers** [sr320/course-fish546-2021#24] [Q&A] [answered]
The reading touched on this a little bit but I am wanting to make sure I understand. The shell is really good for a quick look and general manipulation or cleaning of your data. Further analysis is better done in something like R. 
So, the files that you output from your work in the terminal can then be used in R or in Jupyter? How do Jupyter and R work together? Can they?

**Comment by kubu4** [comment, marked as answer]
Per discussion (just writing it down for posterity) in class today:

> So, the files that you output from your work in the terminal can then be used in R or in Jupyter?

Yep! R and Jupyter (essentially Python) can work with any files you want them to work with. The bigger issues usually comes down to understanding what you want to do with the files and how you need to manipulate them.

> How do Jupyter and R work together? Can they?

They can, but it's a little clunky. Jupyter runs something called "kernels" to allow it to use an underlying programming language. By default, Jupyter runs a Python kernel. You can also install additional kernels (e.g. R) and run those instead of Python. So, if you had the R kernel installed in Jupyter, you would have the option of choosing a Python notebook or an R notebook when you create a new notebook.

To add additional complexity, I think there's a way to have an R kernel available so that you could run individual cells in a Jupyter notebook as R code. So, you could have a notebook with cells where some were run in Python, some in bash, some in R, etc.
